packages/backend/src/managers/modelsManager.ts

```diff
@@ -1,17 +1,53 @@
 import type { LocalModelInfo } from '@shared/src/models/ILocalModelInfo';
 import fs from 'fs';
 import * as path from 'node:path';
+import { type Webview, fs as apiFs } from '@podman-desktop/api';
+import { MSG_NEW_LOCAL_MODELS_STATE } from '@shared/Messages';
+import type { CatalogManager } from './catalogManager';
 
 export class ModelsManager {
-  constructor(private appUserDirectory: string) {}
+  #modelsDir: string;
+  #localModels: Map<string, LocalModelInfo>;
 
-  getLocalModels(): LocalModelInfo[] {
-    const result: LocalModelInfo[] = [];
-    const modelsDir = path.join(this.appUserDirectory, 'models');
-    if (!fs.existsSync(modelsDir)) {
-      return [];
+  constructor(
+    private appUserDirectory: string,
+    private webview: Webview,
+    private catalogManager: CatalogManager,
+  ) {
+    this.#modelsDir = path.join(this.appUserDirectory, 'models');
+    this.#localModels = new Map();
+  }
+
+  async loadLocalModels() {
+    const reloadLocalModels = async () => {
+      this.getLocalModelsFromDisk();
+      const models = this.getModelsInfo();
+      await this.webview.postMessage({
+        id: MSG_NEW_LOCAL_MODELS_STATE,
+        body: models,
+      });
+    };
+    const watcher = apiFs.createFileSystemWatcher(this.#modelsDir);
+    watcher.onDidCreate(reloadLocalModels);
+    watcher.onDidDelete(reloadLocalModels);
+    watcher.onDidChange(reloadLocalModels);
+    // Initialize the local models manually
+    await reloadLocalModels();
+  }
+
+  getModelsInfo() {
```

Comment:
Do we do not display models that are not in the catalog ?

Reply:
Right, we are displaying in the Models list only the models on the disk AND on the catalog, as the catalog is the source for many information (model name, license, etc).

Reply:
I would rather still list the models, and show some "?" or information/warning icon the missing field rather than ignoring them

Reply:
@slemeur what do you think?